### ISSUE TITLE
docker with static port could fail to restart because of port conflict

### ISSUE THREAD

**Issue by camerondavison** [hashicorp/nomad#1191]
### Nomad version

0.3.2
### Operating system and Environment details

CoreOS
### Issue

restarting a docker job with a static port assigned to it can error out.
### Reproduction steps

set a static ip for the port map

```
port "http" {
  static = 1234
}
```
### Nomad Client logs (if appropriate)

```
21/05/16 02:39:10 UTC  Not Restarting  Error was unrecoverable
21/05/16 02:39:10 UTC  Driver Failure  Failed to start container 3a454a67533e25754a8adee1f809760c04e48058fa8f9425b203ad1862d59f33: API error (500): Cannot start container 3a454a67533e25754a8adee1f809760c04e48058fa8f9425b203ad1862d59f33: failed to create endpoint task-name-a547e521-ac69-33b2-fe30-97c12aeed82b on network bridge: Bind for 172.31.31.205:9042 failed: port is already allocated
21/05/16 02:39:10 UTC  Received        Task received by client
```

**Comment by eldondev** [comment]
I also observed this behaviour.

**Comment by a86c6f7964** [comment]
It also marked it as unrecoverable failure, meaning that it did not get rescheduled to a new machine.

**Comment by dadgar** [comment]
So this is issue is due to the fact that when the scheduler sends down the signal to remove the old version and start the new, there is no gating that the new one should be started after the old is torn down. That is the reason there is a port conflict. This is fixable

**Comment by diptanu** [comment]
Fixed via blocking allocations on the same node when we restart an existing allocation.

@a86c6f7964 Can you please see if this actually fixed?

**Comment by grobinson-blockchain** [comment]
I'm still observing this issue with Nomad 0.5.0. Is the fix in this or prior releases?

**Comment by dadgar** [comment]
@grobinson-blockchain Are you observing this when updating a job that has a static port? Can you paste your repro steps. It hasn't been "fixed" in 0.5 but is should be more mitigated

**Comment by camerondavison** [comment]
i have not see this any more
